packages/std/src/entry_points.rs

```diff
@@ -0,0 +1,137 @@
+/// This macro generates the boilerplate required to call into the
+/// contract-specific logic from the entry-points to the WASM module.
+///
+/// It macro should be invoked in a global scope, and the argument to the macro
+/// should be the name of a rust module that is imported in the invocation scope.
+/// The module should export three functions with the following signatures:
+/// ```
+/// # use cosmwasm_std::{
+/// #     Storage, Api, Querier, Extern, Env, StdResult, Binary,
+/// #     InitResult, HandleResult, QueryResult,
+/// # };
+/// #
+/// # type InitMsg = ();
+/// pub fn init<S: Storage, A: Api, Q: Querier>(
+///     deps: &mut Extern<S, A, Q>,
+///     env: Env,
+///     msg: InitMsg,
+/// ) -> InitResult {
+/// #   Ok(Default::default())
+/// }
+///
+/// # type HandleMsg = ();
+/// pub fn handle<S: Storage, A: Api, Q: Querier>(
+///     deps: &mut Extern<S, A, Q>,
+///     env: Env,
+///     msg: HandleMsg,
+/// ) -> HandleResult {
+/// #   Ok(Default::default())
+/// }
+///
+/// # type QueryMsg = ();
+/// pub fn query<S: Storage, A: Api, Q: Querier>(
+///     deps: &Extern<S, A, Q>,
+///     msg: QueryMsg,
+/// ) -> QueryResult {
+/// #   Ok(Binary(Vec::new()))
+/// }
+/// ```
+/// Where `InitMsg`, `HandleMsg`, and `QueryMsg` are types that implement `DeserializeOwned + JsonSchema`
+///
+/// # Example
+///
+/// ```ignore
+/// use contract; // The contract module
+///
+/// cosmwasm_std::entry_points!(contract);
+/// ```
+#[macro_export]
+macro_rules! entry_points {
```

Comment:
Very nice. I just suggest starting this with a verb, like create_*, export_*, connect_* …

Reply:
what would be better, `create_entry_points!()`, or `create_entry_points_for!()`

Reply:
`create_entry_points` is cool I think

Reply:
addressed in 0ae13e8